src/dfcx_scrapi/core/agents.py

```diff
@@ -92,6 +92,47 @@ def get_agent(self, agent_id: str) -> types.Agent:
         response = client.get_agent(request)
 
         return response
+    
+    def get_agent_by_display_name(
+        self, 
+        location_id: str,
```

Comment:
Requiring `location_id` for this type of "user friendly" function seems like a bit of an anti-pattern.
Instead, consider using the GCP Project ID for an arg like `project_id`

The motivation here is that if the user is already using a human readable display name for agent like `My Cool Agent` then it would be cumbersome for them to remember a format like `projects/my-project/locations/us-central1`

Instead, we can do some of the above work for the user under the hood.  
- User provides human readable Agent Display Name = `My Cool Agent`
- User provides human readable GCP Project ID = `my-project`
- We handle the rest

Check out `dfcx_scrapi.soop.project.py.list_intents` for some intuition.  
Or simply import that and use it as it should be functional.
That particular file is still in beta/development but we can look to preserve dependencies.

Reply:
So you're thinking based on just the the `GCP_PROJECT_ID` we can find the `LOCATION_ID` by searching in all locations, for any agents matching `display_name` ?

`projects/<GCP PROJECT ID>/locations/<LOCATION ID>`

There could be a risk of duplicates? tho it's rare people would duplicate across regions? maybe someone trying to do global multi-region deploy themselves. backups in a different GCP project maybe common though and this should avoid a name collision in that case since `GCP-PJID` is required.

We could add a fuzzy finder too, depending on where this is being used, eg just as a one off search tool, vs in an unattended cron task.

Reply:
Would it be simplest, safest, and most user friendly to simply add `LOCATION_ID` as a third parameter for the function?

Reply:
Apologies for the back and forth. So there's a couple of things per our discussion via chat.

1. Dialogflow CX console doesn't enforce unique agent Display Names at a Project ID level, but only a Project ID + Region ID combination level. (i.e. "My Cool Agent" can exist as an agent name in both `us-central1` and `us-east1` simultaneously) Seems like not that best UI/UX given these all appear next to each other on the Agent List page but 🤷 . So that said, we should likely opt for a couple of scenarios to allow for user flexibility:
    - Accept `display_name` and `location_id` as args
    - Accept `display_name`, `project_id`, `region_id` as args
While the former covers the latter, it also includes the more rigid structure of knowing the path format for `location_id`. The latter would be a more user friendly approach and still provide the same end result.

2. I mentioned previously to use `soop.project.list_agents` but later realized this would cause some circular import logic. I went ahead and added the logic to gather all agents across all regions directly in `core.agents.list_agents` now, so you can simply use it directly in the class. It will now accept either a `location_id` or a `project_id` depending on your use case/needs.